base_location/models/company.py

```diff
@@ -19,31 +20,26 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 #
-from openerp.osv import orm, fields
+from openerp import models, fields, api
 
 
-class ResCompany(orm.Model):
+class ResCompany(models.Model):
 
     _inherit = 'res.company'
 
-    def on_change_city(self, cr, uid, ids, zip_id, context=None):
-        result = {}
-        if context is None:
-            context = {}
-        if zip_id:
-            bzip = self.pool['res.better.zip'].browse(
-                cr, uid, zip_id, context=context)
-            result = {'value': {
-                'zip': bzip.name,
-                'country_id': bzip.country_id.id if bzip.country_id else False,
-                'city': bzip.city,
-                'state_id': bzip.state_id.id if bzip.state_id else False
-                }
-            }
-        return result
+    @api.multi
```

Comment:
@alejandrosantana It is only more efficient if you use `self.write()` outside your for loop in an `@api.multi` when setting values. But this doesn't apply to onchanges as it only modifies the cache.